### PR TITLE
Data preservation

### PR DESCRIPTION
**Pull request by tomdan-ai**
### The team implemented the following changes:

 - Deserialized JSON data as seen in Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
 - On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and                  
     named accordingly (`rentals.json`, `people.json`, `classroom.json`, `rental.json` )
 - On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -9,7 +16,7 @@ def handle_menu_choice(choice, app)
     5 => { description: 'Create a book', action: -> { app.create_book } },
     6 => { description: 'Create a rental', action: -> { app.create_rental } },
     7 => { description: 'List rentals for a person', action: -> { app.list_rentals_for_person } },
-    8 => { description: 'Quit', action: -> { exit_app } }
+    8 => { description: 'Quit', action: -> { exit_app(app) } }
```
Comment (DuaneDave) [inline main.rb:19]: You have done a good job in setting up the console, however, when I quit the app, I get an error thrown on the console and I think it's with regards to the book title. Here is a screenshot to show the error I got. Consider addressing this in order to make the persistence of user data work smoothly when there is nothing yet saved.

error

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @tomdan-ai and @ernestkorir 👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Your console app design looks great

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (tomdan-ai) [reply main.rb:19]: The error you have doesnt exist in mine, I believe it may be caused by an external issue

**Review by addod19 — APPROVED** [review]
### Hi @tomdan-ai and @ernestkorir, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] The app quits as expected :rocket: 
Screenshot from 2023-07-04 20-08-29


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**